web/autolinker.js

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2025 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createValidAbsoluteUrl, Util } from "../src/shared/util.js";
+import { getOriginalIndex, normalize } from "./pdf_find_controller.js";
+
+class Autolinker {
+  static #urlRegex =
+    /\b(?:https?:\/\/|mailto:|www\.)(?:[[\S--\[]--\p{P}]|\/|[\p{P}--\[]+[[\S--\[]--\p{P}])+|\b[\S--@]+@[\S--.]+.[[\S--\[]--\p{P}]{2,}/gmv; // Regex can be tested and verified at https://regex101.com/r/zuMeQX/3.
+
+  static #addLinkAnnotations(url, index, length, pdfPageView) {
+    const convertedMatch = pdfPageView._textHighlighter._convertMatches(
```

Comment:
If we are using `_convertMatches` outside of the `_textHighlighter` class, we should probably remove the `_` from its name.

Reply:
What about `_textHighlighter` itself?

Reply:
No one seconded this comment, so I'm keeping things as-is for now.